File: src/main/java/com/ultreon/mods/screenshotmanager/client/ScreenshotCache.java
package com.ultreon.mods.screenshotmanager.client;

import com.mojang.blaze3d.platform.NativeImage;
import net.minecraft.client.renderer.texture.DynamicTexture;

import java.io.File;
import java.util.HashMap;
import java.util.Map;

public class ScreenshotCache {
    private static final ScreenshotCache INSTANCE = new ScreenshotCache();
    private final Map<File, ScreenshotData> cache = new HashMap<>();

    public ScreenshotData cache(File name, DynamicTexture texture) {
        NativeImage pixels = texture.getPixels();
        ScreenshotData data;
        if (pixels == null) {
            data = new ScreenshotData(0, 0);
        } else {
            data = new ScreenshotData(pixels.getWidth(), pixels.getHeight());
        }
        this.cache.put(name, data);

        return data;
    }

    public ScreenshotData get(File file) {
        return cache.get(file);
    }

    public ScreenshotData getOrDefault(File file) {
        return cache.getOrDefault(file, new ScreenshotData(16, 16));
    }

    public static ScreenshotCache get() {
        return INSTANCE;
    }

    public boolean isCached(File file) {
        return cache.containsKey(file);
    }
}

File: src/main/java/com/ultreon/mods/screenshotmanager/client/ScreenshotCache.kt
package com.ultreon.mods.screenshotmanager.client

import com.badlogic.gdx.graphics.Texture
import java.io.File

object ScreenshotCache {
    private val cache: MutableMap<File, ScreenshotData> = HashMap()
    fun cache(name: File, texture: Texture): ScreenshotData {
        val pixels = texture.isManaged
        val data: ScreenshotData = if (pixels == null) {
            ScreenshotData(0, 0)
        } else {
            ScreenshotData(texture.textureData.width, texture.textureData.height)
        }
        cache[name] = data
        return data
    }

    operator fun get(file: File): ScreenshotData? {
        return cache[file]
    }

    @Deprecated("Replaced with getOrEmpty.", ReplaceWith(
        "getOrEmpty(file)",
        "com.ultreon.mods.screenshotmanager.client.ScreenshotCache.getOrEmpty"
    ))
    fun getOrDefault(file: File): ScreenshotData {
        return getOrEmpty(file)
    }

    fun getOrEmpty(file: File): ScreenshotData {
        return cache.getOrDefault(file, ScreenshotData(16, 16))
    }

    fun isCached(file: File): Boolean {
        return cache.containsKey(file)
    }

    @Deprecated("Object class now.", ReplaceWith("ScreenshotCache"))
    fun get(): ScreenshotCache {
        return ScreenshotCache
    }
}
